internal/backend/db/dbgorm/ownership.go

```diff
@@ -154,72 +145,72 @@ func isUserEntity(db *gorm.DB, user *scheme.User, ids ...sdktypes.UUID) error {
+	return gdb.isUserEntity(uid, ids...)
 }
 
 // REVIEW: this is probably the simplest possible way (e.g. with entity as string).
 // We could also use generics, TableName, interface to find ID column,  etc..
 
 // join with user ownership on entity
-func joinUserEntity(db *gorm.DB, entity string, userID sdktypes.UUID) *gorm.DB {
+func joinUserEntity(db *gorm.DB, entity string, uid string) *gorm.DB {
 	tableName := entity + string("s")
 	joinExpr := fmt.Sprintf("JOIN ownerships ON ownerships.entity_id = %s.%s_id", tableName, entity)
-	return db.Table(tableName).Joins(joinExpr).Where("ownerships.user_id = ?", userID)
+	return db.Table(tableName).Joins(joinExpr).Where("ownerships.user_id = ?", uid)
 }
 
 // gorm user+entity scope
-func withUserEntity(gdb *gormdb, entity string, user *scheme.User) func(*gorm.DB) *gorm.DB {
+func withUserEntity(gdb *gormdb, entity string, uid string) func(*gorm.DB) *gorm.DB {
 	return func(db *gorm.DB) *gorm.DB {
-		return gdb.owner.JoinUserEntity(db, entity, user)
+		return gdb.owner.JoinUserEntity(db, entity, uid)
 	}
 }
 
 // gormdb user+entity scoped godm db + logging
 func (gdb *gormdb) withUserEntity(ctx context.Context, entity string) *gorm.DB {
-	user, _ := userFromContext(ctx) // NOTE: ignore possible error
+	user, _ := userIDFromContext(ctx) // NOTE: ignore possible error
 	return gdb.owner.JoinUserEntity(gdb.db.WithContext(ctx), entity, user)
 }
 
 type OwnershipChecker interface {
-	IsUserEntity(db *gorm.DB, user *scheme.User, ids ...sdktypes.UUID) error
-	JoinUserEntity(db *gorm.DB, entity string, user *scheme.User) *gorm.DB
+	IsUserEntity(db *gorm.DB, user string, ids ...sdktypes.UUID) error
+	JoinUserEntity(db *gorm.DB, entity string, user string) *gorm.DB
 }
 
 type UsersOwnershipChecker struct {
 	z *zap.Logger
 }
 
-func (c *UsersOwnershipChecker) IsUserEntity(db *gorm.DB, user *scheme.User, ids ...sdktypes.UUID) error {
-	c.z.Debug("isUserEntity", zap.Any("entityIDs", ids), zap.Any("user", user))
-	return isUserEntity(db, user, ids...)
+func (c *UsersOwnershipChecker) IsUserEntity(db *gorm.DB, uid string, ids ...sdktypes.UUID) error {
+	c.z.Debug("isUserEntity", zap.Any("entityIDs", ids), zap.Any("uid", uid))
+	return isUserEntity(db, uid, ids...)
 }
 
-func (c *UsersOwnershipChecker) JoinUserEntity(db *gorm.DB, entity string, user *scheme.User) *gorm.DB {
-	c.z.Debug("withUser", zap.String("entity", entity), zap.Any("user", user))
-	return joinUserEntity(db, entity, user.UserID)
+func (c *UsersOwnershipChecker) JoinUserEntity(db *gorm.DB, entity string, uid string) *gorm.DB {
+	c.z.Debug("withUser", zap.String("entity", entity), zap.Any("uid", uid))
+	return joinUserEntity(db, entity, uid)
 }
 
 type PermissiveOwnershipChecker struct {
 	z *zap.Logger
 }
 
-func (c *PermissiveOwnershipChecker) IsUserEntity(db *gorm.DB, user *scheme.User, ids ...sdktypes.UUID) error {
+func (c *PermissiveOwnershipChecker) IsUserEntity(db *gorm.DB, uid string, ids ...sdktypes.UUID) error {
 	return nil
```

Comment:
What is this function?

Reply:
@ig-ra i'm not sure?

Reply:
function that returns whether provided IDs are belong/accessible by the given user